### PR TITLE
Add patient enrichments setting to mutation and cna

### PR DESCRIPTION
**Pull request by adamabeshouse**
Adds ability to query patient-level enrichments in mutation and cna enrichments tabs, in results view and group comparison

image

### CODE REVIEW

File: src/pages/groupComparison/GroupComparisonStore.ts [hunk 1]
```diff
@@ -111,6 +113,11 @@ export default class GroupComparisonStore {
         return param || OverlapStrategy.EXCLUDE;
     }
 
+    @autobind
+    @action public setUsingPatientLevelEnrichments(e:boolean) {
+        this.usingPatientLevelEnrichments = e;
```
Comment (alisman) [inline src/pages/groupComparison/GroupComparisonStore.ts]: usePatientLevelEnrichments?

**Comment by adamabeshouse** [comment]
coocurrence pattern in copy number amp table is broken

image

https://beta.cbioportal.org/results/enrichments?Z_SCORE_THRESHOLD=2.0&cancer_study_id=coadread_tcga_pub&cancer_study_list=coadread_tcga_pub&case_ids=&case_set_id=coadread_tcga_pub_nonhypermut&gene_list=KRAS%20NRAS%20BRAF&gene_set_choice=user-defined-list&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=coadread_tcga_pub_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=coadread_tcga_pub_mutations

**Comment by alisman** [comment]
The new dropdown seems greyed out to begin with.  (unlike enriched groups dropdown to the left in which the arrow is full black).   There are other inconsistencies about the style of the dropdown.  I know it's going to be a PITA to figure out but would be nice to get em the same. 
image

**Comment by tmazor** [comment]
I am so excited about this!!

A few thoughts:

1) Can the table column tooltips change based on the patient vs sample selection? Some tooltips refer to samples but should refer to patients when patient-level is selected (Altered group/Unaltered group/group names and Co-occurrence pattern)

2) I second Aaron's request to make the Patient/Sample dropdown look the same as the other dropdown.

3) If I change Copy-number to Patient-Level, an error appears: https://cbioportal-frontend-pr-2534.herokuapp.com/comparison/cna?sessionId=5d2f8546e4b0ab413787bf8d

4) There's something off to me about the layout here:
image
I think it's that the significant checkbox seems out of place in between the two dropdown menus. Maybe this can be re-ordered? Or put some separation between the significant checkbox and the patient/sample dropdown? @alisman any suggestions?

**Comment by adamabeshouse** [comment]
@tmazor @alisman fixed almost all of your comments, just remains @tmazor's #4. 

@tmazor your #3 is just caused by master API being out of date with beta. Looks like you need to use beta.cbioportal.org with `localStorage.heroku = "cbioportal-frontend-pr-2534"`

**Comment by adamabeshouse** [comment]
@tmazor @jjgao  @alisman updates pushed

**Comment by alisman** [comment]
@adamabeshouse awesome!  one more super annoying request: if you make the height of the control 36px instead of 33.5, the centering is perfect.

**Comment by adamabeshouse** [comment]
@alisman done!

**Comment by tmazor** [comment]
Yup, using beta fixes that CNA issue, and everything else is looking good @adamabeshouse !

Just one small and super nitpicky thing - I think there's still a slight difference in the text in the two dropdowns - the "Select enriched groups" text looks stronger to me than "Sample/Patient-level enrichments". I can't lay my finger on exactly what it is, but there's something different (the font weight maybe?). For the record, I prefer the way the "patient/sample enrichments" text looks, so if possible, can the "select enrichments group" text be changed?

**Comment by adamabeshouse** [comment]
@tmazor seems that the only difference is the colors. I updated it to use the same colors, let me know if you like it.

**Comment by tmazor** [comment]
@adamabeshouse did the color change get deployed to heroku? The text still looks different and when I inspect the elements, the text is two different colors still.

**Comment by adamabeshouse** [comment]
@tmazor yeah its showing up for me

**Comment by jjgao** [comment]
@adamabeshouse quick question: for [this query](https://cbioportal-frontend-pr-2534.herokuapp.com/comparison/overlap?sessionId=5cf89323e4b0ab413787436c) with overlapping patients but no overlapping samples, whencomparing samples for Mutations, are all samples included? When switching to patients, are the overlapping patients excluded?

I find the message below confusing...

image

**Comment by adamabeshouse** [comment]
@jjgao you're right, that message is completely confusing. not related to this issue... but confusing :). I'll log it

**Comment by jjgao** [comment]
@adamabeshouse ok, thanks! But this question still needs an answer for this PR: for [this query](https://cbioportal-frontend-pr-2534.herokuapp.com/comparison/overlap?sessionId=5cf89323e4b0ab413787436c) with overlapping patients but no overlapping samples, when comparing samples for Mutations, are all samples included? When switching to patients, are the overlapping patients excluded?

**Comment by adamabeshouse** [comment]
we're still missing one screenshot test, because circleCI is going to rc instead of beta, and the test is breaking

**Comment by adamabeshouse** [comment]
@jjgao overlap is computed and excluded separately for samples and patients

**Comment by adamabeshouse** [comment]
@jjgao just pushed an update and it should be working properly now

**Comment by jjgao** [comment]
@adamabeshouse cool! Is it possible to add the patient/sample option to the url?

**Comment by adamabeshouse** [comment]
done @jjgao

**Pull request merged**